examples/data/interpolation_evalscript_cdse.js

```diff
@@ -0,0 +1,177 @@
+var n_all = 0;
+
+function interval_search(x, arr) {
+  let start_idx = 0,  end_idx = arr.length - 2;
+
+  // Iterate while start not meets end
+  while (start_idx <= end_idx) {
+    // Find the mid index
+    let mid_idx = (start_idx + end_idx) >> 1;
+
+    // If element is present at mid, return True
+    if (arr[mid_idx] <= x && x < arr[mid_idx + 1]) {
+      return mid_idx;
+    }
+    // Else look in left or right half accordingly
+    else if (arr[mid_idx + 1] <= x) start_idx = mid_idx + 1;
+    else end_idx = mid_idx - 1;
+  }
+  if (x == arr[arr.length-1]){
+    return arr.length-2;
+  }
+  return undefined;
+}
+
+function linearInterpolation(x, x0, y0, x1, y1, no_data_value=NaN) {
+  if (x < x0 || x > x1) {
+    return no_data_value;
+  }
+  var a = (y1 - y0) / (x1 - x0);
+  var b = -a * x0 + y0;
+  return a * x + b;
+}
+
+function lininterp(x_arr, xp_arr, fp_arr, no_data_value=NaN) {
+  results = [];
+  data_mask = [];
+  xp_arr_idx = 0;
+  for (var i=0; i<x_arr.length; i++) {
+    var x = x_arr[i];
+    n_all+=1;
+    interval = interval_search(x, xp_arr);
+    if (interval === undefined) {
+      data_mask.push(0);
+      results.push(no_data_value);
+      continue;
+    }
+    data_mask.push(1);
+    n_valid+=1;
+    results.push(
+      linearInterpolation(
+        x,
+        xp_arr[interval],
+        fp_arr[interval],
+        xp_arr[interval+1],
+        fp_arr[interval+1],
+        no_data_value
+      )
+    );
+  }
+  return [results, data_mask];
+}
+
+function interpolated_index(index_a, index_b) {
+  // Calculates the index for all bands in array
+  var index_data = [];
+  for (var i = 0; i < index_a.length; i++){
+     // UINT index returned
+     let ind = (index_a[i] - index_b[i]) / (index_a[i] + index_b[i]);
+     index_data.push(ind * 10000 + 10000);
+  }
+  return index_data
+}
+
+function increase(original_date, period, period_unit) {
+    date = new Date(original_date)
+    switch (period_unit) {
+        case 'millisecond':
+            return new Date(date.setMilliseconds(date.getMilliseconds()+period));
+        case 'second':
+            return new Date(date.setSeconds(date.getSeconds()+period));
+        case 'minute':
+            return new Date(date.setMinutes(date.getMinutes()+period));
+        case 'hour':
+            return new Date(date.setHours(date.getHours()+period));
+        case 'day':
+            return new Date(date.setDate(date.getDate()+period));
+        case 'month':
+            return new Date(date.setMonth(date.getMonth()+period));
+        default:
+            return undefined
+    }
+}
+
+function sample_timestamps(start, end, period, period_unit) {
+    var cDate = new Date(start);
+    var sampled_dates = []
+    while (cDate < end) {
+        sampled_dates.push(cDate);
+        cDate = increase(cDate, period, period_unit);
+    }
+    return sampled_dates;
+}
+
+function is_valid(smp) {
+  // Check if the sample is valid (i.e. contains no clouds or snow)
+  let scl = smp.SCL;
```

Comment:
Give me context, these cdse evalscripts are included because some bands are different?

Reply:
Yes, the example was using CLM which does not exist in CDSE